documentation/design-docs/ipc-protocol.md

```diff
@@ -0,0 +1,183 @@
+# Event Pipe
```

Comment:
The IPC mechanism isn't only for EventPipe tracing, it is for any tool that wants the runtime to do something. Two other use cases we may have before 3.0 ships:
a) attach a profiler
b) take a dump